src/java.desktop/share/classes/javax/swing/JTable.java

```diff
@@ -1266,6 +1266,12 @@ public void setAutoResizeMode(int mode) {
             autoResizeMode = mode;
             resizeAndRepaint();
             if (tableHeader != null) {
+                if (mode == JTable.AUTO_RESIZE_LAST_COLUMN) {
+                    if (columnModel.getColumnCount() > 0) {
```

Comment:
Conditions can be combined

`if (mode == JTable.AUTO_RESIZE_LAST_COLUMN
     && columnModel.getColumnCount() > 0) {`